tests/components/template/test_init.py

```diff
@@ -253,5 +257,50 @@ async def test_reloadable_multiple_platforms(hass):
     assert float(hass.states.get("sensor.combined_sensor_energy_usage").state) == 0
 
 
+async def test_reload_sensors_that_reference_other_template_sensors(hass):
+    """Test that we can reload sensor that reference other template sensors."""
+
+    await async_setup_component(
+        hass,
+        "sensor",
+        {
+            "sensor": {
+                "platform": DOMAIN,
+                "sensors": {
+                    "state": {"value_template": "{{ 1 }}"},
+                },
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    yaml_path = path.join(
+        _get_fixtures_base_path(),
+        "fixtures",
+        "template/ref_configuration.yaml",
+    )
+    with patch.object(config, "YAML_CONFIG_FILE", yaml_path):
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_RELOAD,
+            {},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) == 3
+    await hass.async_block_till_done()
+
+    next_time = dt_util.utcnow() + timedelta(seconds=1.2)
+    with patch(
+        "homeassistant.helpers.ratelimit.dt_util.utcnow", return_value=next_time
+    ):
+        async_fire_time_changed(hass, next_time)
+        await hass.async_block_till_done()
+
+    assert hass.states.get("sensor.test1").state == "3"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6271139)